runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -1194,10 +1194,11 @@ TR::Register *J9::X86::TreeEvaluator::asynccheckEvaluator(TR::Node *node, TR::Co
       {
       TR::TreeEvaluator::asyncGCMapCheckPatching(node, cg, snippetLabel);
       }
-   else if (secondChild->getOpCode().isLoadConst())
+   else
       {
-      TR::MemoryReference *mr = generateX86MemoryReference(compareNode->getFirstChild(), cg);
+      TR_ASSERT(secondChild->getOpCode().isLoadConst(), "unrecognized asynccheck test: special async check value is not a constant");
```

Comment:
Actually - this needs to be fatal - previously we would return NULL without generating the check - now we would try to generate the checking code assuming a const load if the assert were not enabled.

Reply:
Good point. I've updated it to `TR_ASSERT_FATAL`.